src/undate/converters/calendars/hijri/converter.py

```
@@ -0,0 +1,48 @@
+
+        # parse the input string, then transform to undate object
+        try:
+            # parse the string with our Hijri date parser
+            parsetree = hijri_parser.parse(value)
+            # transform the parse tree into an undate or undate interval
+            undate_obj = self.transformer.transform(parsetree)
+            # set the original date as a label, with the calendar name
+            undate_obj.label = f"{value} {self.calendar_name}"
+            return undate_obj
+        except UnexpectedCharacters:
+            raise ValueError("Could not parse '%s' as a Hijri date" % value)
```

Ruff (0.7.0)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)